arch/rv32i/src/machine_timer.rs

```diff
@@ -46,20 +39,27 @@ impl MachineTimer<'_> {
     }
 
     fn disable_machine_timer(&self) {
-        // Disable by setting the mtimecmp register to its max value, which
-        // we will never hit.
-        self.registers
-            .mtimecmp
-            .write(MTimeCmp::MTIMECMP.val(0xFFFF_FFFF_FFFF_FFFF));
+        // Disable by setting the mtimecmp register to its max value.
+        // In theory we won't hit it if we start at 0, but it's ridiculous that
+        // mtimer has no enable/disable mechanism.
+        self.registers.compare_high.set(0xFFFF_FFFF);
+        self.registers.compare_low.set(0xFFFF_FFFF);
```

Comment:
I notice that in `set_alarm()` we set the interrupt enable bit for the mtimer, could we not just clear that bit here?